omnigibson/prims/xform_prim.py

```diff
@@ -300,7 +284,18 @@ def set_local_pose(self, position=None, orientation=None):
             else:
                 rotq = lazy.pxr.Gf.Quatd(*orientation)
             xform_op.Set(rotq)
+        PoseAPI.invalidate()
         BoundingBoxAPI.clear()
+        if gm.ENABLE_FLATCACHE:
+            # If flatcache is on, make sure the USD local pose is synced to the fabric local pose.
+            # Ideally we should call usdrt's set local pose directly, but there is no such API.
+            # The only available API is SetLocalXformFromUsd, so we update USD first, and then sync to fabric.
+            xformable_prim = lazy.usdrt.Rt.Xformable(lazy.omni.isaac.core.utils.prims.get_prim_at_path(self.prim_path, fabric=True))
+            # We also need to clear the fabric's world pose (potentially set by physx) to force fabric to use
+            # the local pose we just set, because world pose takes precendence over local pose.
+            # if xformable_prim.HasWorldXform():
+            assert not xformable_prim.HasWorldXform(), "Fabric's world pose is set for a non-rigid prim which is unexpected. Please report this."
```

Comment:
should be explicitly say "xformprim" instead of "non-rigid prim"? Or are there other cases where this applies?

Reply:
So the gist of it is that for anything that has a physx counterpart (articulation, rigidbody) there will be a worldxform but that's OK because this code will never get called for those objects. It's only the non-physx stuff that this gets called for, and for those there should never be a worldxform or the rest of this code wont work